src/reducers/index.js

```diff
@@ -1500,4 +1502,28 @@ export const onUpdateNetworkSettings = (state, { payload }) => {
   }
 }
 
+/**
+ * Handle successful download of unregistered token details
+ * @param {Object} state Current state
+ * @param {Object} action Action with token details
+ * @param {Object} action.payload.tokens Object with token details
+ */
+export const onUnregisteredTokensDownloadSuccess = (state, action) => {
+  const { tokens } = action.payload;
+  const newTokens = Object.values(tokens).map(token => ({
+    uid: token.uid,
+    name: token.name,
+    symbol: token.symbol
+  }));
+
+  return {
+    ...state,
+    tokens: [...state.tokens, ...newTokens],
+    tokenMetadata: {
+      ...state.tokenMetadata,
+      ...tokens
+    }
```

Comment:
I didn't understand this. To update the token metadata you should call the metadata API.

Reply:
This was indeed wrong and not fetching the actual token metadata.

Refactored the `requestUnregisteredTokensDownload` to actually dispatch the action to fetch tokens metadata

Since this code was copied from the wallet-mobile, I'll check if we're doing the same there